### ISSUE TITLE
DraggableScrollableSheet not working as expected on Desktop Web

### ISSUE THREAD

**Issue by SeriousMonk** [flutter/flutter#101903]
DraggableScrollableSheet works perfectly on mobile and mobile browsers, however on web it doesn't.
On web it only works as expected if user drags the list with the mouse; if the user uses the scroll wheel (much more natural behavior on desktop) the DraggableScrollableSheet doesn't expand but stays fixed at its initialChildSize.
DraggableScrollableSheet should expand and collapse even if the user uses the scroll wheel to scroll through the list.

**Comment by maheshmnj** [comment]
Hi @SeriousMonk, Thanks for filing the issue. Labeling this issue as a proposal to allow DraggableScrollableSheeet to consume mouse scroll events along with drag gestures.

cc: @Piinks

**Comment by Piinks** [comment]
I am not sure I understand the request. Can you please provide a minimal code sample, with instructions on how to produce the expected behavior and what the actual behavior is?

**Comment by SeriousMonk** [comment]
On web the draggablescrollablesheet does not expand to maxChildHeight if the scrollable inside is scrolled using the mouse wheel; instead it stays at fixed at initialChildHeight.
It happens with any implementation, even in one provided by you in another issue.

**Comment by Piinks** [comment]
> even in one provided by you in another issue.

This is not actionable.

Please provide the requested information before we can address the issue.

**Comment by SeriousMonk** [comment]
Sorry, here is an example to reproduce the behavior I'm talking about.
```dart
import 'dart:ui';

import 'package:flutter/material.dart';

void main() => runApp(MyApp());

class MyCustomScrollBehavior extends MaterialScrollBehavior {
  // Override behavior methods and getters like dragDevices
  @override
  Set<PointerDeviceKind> get dragDevices => {
    PointerDeviceKind.touch,
    PointerDeviceKind.mouse,
    // etc.
  };
}

class MyApp extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      scrollBehavior: MyCustomScrollBehavior(),
      title: 'Flutter Hello World',
      theme: ThemeData(
        primarySwatch: Colors.blue,
      ),
      home: const MyHomePage(title: 'Flutter Demo Home Page'),
    );
  }
}

class MyHomePage extends StatelessWidget {
  final String title;

  const MyHomePage({this.title = "Home Page"});

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(
        title: Text(title),
      ),
      body: const Center(
        child: Text(
          'Hello, World!',
        ),
      ),
      floatingActionButton: FloatingActionButton(
        onPressed: () => showModalBottomSheet(
          context: context,
          isScrollControlled: true,
          builder: (_) => DraggableScrollableSheet(
            expand: false,
            builder: (_, ScrollController scrollController) {
              return ListView(
                controller: scrollController,
                children: [for (var i = 0; i < 250; i++) Text("Item: $i")],
              );
            },
          ),
          shape: const RoundedRectangleBorder(
              borderRadius: BorderRadius.vertical(top: Radius.circular(20))),
        ),
        tooltip: "Open Sheet!",
        child: const Icon(Icons.add),
      ),
    );
  }
}
```

Obtained behavior:
1)I scroll using the scroll wheel
2)The DraggableScrollableSheet remains at initialChildSize (0.5 in the example) 
3)The inner listview scrolls inside

The expected behavior would be the following:
1) I scroll using the scroll wheel
2) The DraggableScrollableSheet expands following the scroll until it reaches maxChildSize (1.0 in the example)
3)Only once the DraggableScrollableSheet is at maxChildSize will the inner scrollable scroll like a regular listview

**Comment by 12people** [comment]
The issue seems to pertain to desktop builds too (at least for macOS).

The issue is described quite nicely in this [StackOverflow question](https://stackoverflow.com/questions/69930470/flutter-desktop-web-draggablescrollablesheet-scrolling-problem):

> It works perfectly fine and as expected when opened from mobile device: the DraggableScrollableSheet gets opened, takes 50% of the screen height, and then the ListView inside it may be scrolled along with Sheet itself, so it will take from 0% to 100% of the screen.
> However, if opened in desktop browser, the DraggableScrollableSheet just refuses to be scrolled together with ListView on mouse wheel rotation. It just sticks to 50% and can only be closed and never expanded!

The question itself also includes a code example, in case it helps.

**Comment by maheshmnj** [comment]
> I am not sure I understand the request. Can you please provide a minimal code sample, with instructions on how to produce the expected behavior and what the actual behavior is?

Currently `DraggableScrollableSheet` only supports drag Gestures even on the Desktop, This issue is proposing to support mouse scroll events to be used to scroll the `DraggableScrollableSheet`, similar to how ListView reacts to mouse events.

**Comment by Piinks** [comment]
Thanks for clarifying! So you would like the mouse scroll to initiate what would normally be the drag gesture? Would that be confusing, since it is a draggable sheet? I think it is meant to be dragged, hence the name, but we might be able to explore other behaviors. I have not worked on this widget, so other folks may know better than I do. :)

Do you have any examples of this behavior from existing apps or websites? It would be helpful to see if this is a common UI pattern and how it typically behaves.

It looks like `_DraggableScrollableSheetScrollPosition` does not implement `pointerScroll`. If it is possible to support this behavior, that would likely be the place to start.

**Comment by SeriousMonk** [comment]
I think that when a user is using a mouse, the most natural way of scrolling is with the mouse wheel. Dragging, on the other hand, comes natural on touch devices.

**Comment by Piinks** [comment]
Right, totally! And the scrollable content does scroll, it's the drag to expand functionality that does not respond to a mouse scroll action.
That makes me wonder if using this widget is not what is wanted on desktop. Some concrete examples of how this type of UI component behaves on desktop with a mouse outside of Flutter will be helpful in determining how to update the behavior.

**Comment by mazino-dev** [comment]
It would be nice if the widget behaves the same for desktop.

**Comment by Piinks** [comment]
> It would be nice if the widget behaves the same for desktop.

Hey @mazino-dev, can you clarify? I have been gathering info on this. On desktop, does it not drag? That would mean it is behaving the same way from what I can tell. On mobile you have to drag. :)

**Comment by 12people** [comment]
@Piinks No — neither scrolling nor dragging does anything on the desktop (at least not on macOS). In fact, there doesn't seem a way to ever achieve the maxChildSize — the sheet just stays at its initialChildSize on the desktop.

**Comment by mazino-dev** [comment]
12 people are correct. DraggableScrollableSheet leaves out the Draggable part on desktop or maybe the only thing it dragged was my pride in making it work.Before flutter, we were spoiled by [this](https://gorhom.github.io/react-native-bottom-sheet/modal/) in the React Native land, maybe DraggableScrollableSheet can look upto its cousin for what customizations scrolled but not so dragged people like me expect.

**Comment by maheshmnj** [comment]
> @Piinks No — neither scrolling nor dragging does anything on the desktop (at least not on macOS). In fact, there doesn't seem a way to ever achieve the maxChildSize — the sheet just stays at its initialChildSize on the desktop.

I can confirm this, DraggableScrollableSheet is broken on the desktop, the only way to make it work is by overriding the dragBehavior as shown here https://github.com/flutter/flutter/issues/101903#issuecomment-1099701711

**Comment by 12people** [comment]
It seems like it works now, BUT only with scrolling.

Dragging a partially shown sheet up doesn't do anything, but dragging it down hides the sheet. It's oddly inconsistent.

**Comment by Piinks** [comment]
@thkim1011 has been looking at this recently

**Comment by huycozy** [comment]
While working on another issue, I found this issue and started doing some tests on devices, environments, scroll behaviors.

### 1. Default scroll behavior

Since PR https://github.com/flutter/flutter/pull/81569 has been merged, the vertical drag gesture is disabled by default on Desktop and Web platforms; to align with the natural behavior found in other desktop applications, see [Default drag scrolling devices](https://docs.flutter.dev/release/breaking-changes/default-scroll-behavior-drag).

![default](https://github.com/flutter/flutter/assets/104349824/da4d18e2-c5eb-46d0-8705-9337a3e8524f)

<details>
<summary>Sample code (default scroll behavior)</summary>

```dart
import 'package:flutter/material.dart';

void main() {
  runApp(const MaterialApp(
    home: HomePage(),
  ));
}

class HomePage extends StatelessWidget {
  const HomePage({super.key});

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(
        title: const Text('DraggableScrollableSheet'),
      ),
      body: SizedBox.expand(
        child: DraggableScrollableSheet(
          builder: (BuildContext context, ScrollController scrollController) {
            return Container(
              color: Colors.blue[100],
              child: ListView.builder(
                controller: scrollController,
                itemCount: 25,
                itemBuilder: (BuildContext context, int index) {
                  return ListTile(title: Text('Item $index'));
                },
              ),
            );
          },
        ),
      ),
    );
  }
}
```
</details>

### 2. Using a custom scroll behavior

This is for anyone who still desires to activate drag gestures on Desktop and Web platforms (OP mentioned above)

![custom](https://github.com/flutter/flutter/assets/104349824/5a799e4c-a454-40d1-9bb5-a9ac782f08f9)

<details>
<summary>Sample code (use a custom scroll behavior)</summary>

```dart
import 'package:flutter/gestures.dart';
import 'package:flutter/material.dart';

void main() {
  runApp(const MaterialApp(
    home: HomePage(),
  ));
}

class HomePage extends StatelessWidget {
  const HomePage({super.key});

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(
        title: const Text('DraggableScrollableSheet'),
      ),
      body: SizedBox.expand(
        child: DraggableScrollableSheet(
          builder: (BuildContext context, ScrollController scrollController) {
            return ScrollConfiguration(
              behavior: ScrollConfiguration.of(context).copyWith(dragDevices: {
            		PointerDeviceKind.touch,
            		PointerDeviceKind.mouse,
            		PointerDeviceKind.trackpad,
            	}),
              child: Container(
                color: Colors.blue[100],
                child: ListView.builder(
                  controller: scrollController,
                  itemCount: 25,
                  itemBuilder: (BuildContext context, int index) {
                    return ListTile(title: Text('Item $index'));
                  },
                ),
              ),
            );
          },
        ),
      ),
    );
  }
}
```
</details>

<details>
<summary>flutter doctor -v (stable and master)</summary>

```bash
[✓] Flutter (Channel stable, 3.13.8, on macOS 14.0 23A344 darwin-x64, locale en-VN)
    • Flutter version 3.13.8 on channel stable at /Users/huynq/Documents/GitHub/flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 6c4930c4ac (11 hours ago), 2023-10-18 10:57:55 -0500
    • Engine revision 767d8c75e8
    • Dart version 3.1.4
    • DevTools version 2.25.0

[✓] Android toolchain - develop for Android devices (Android SDK version 32.0.0)
    • Android SDK at /Users/huynq/Library/Android/sdk
    • Platform android-34, build-tools 32.0.0
    • ANDROID_HOME = /Users/huynq/Library/Android/sdk
    • Java binary at: /Applications/Android Studio Giraffe.app/Contents/jbr/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 17.0.6+0-17.0.6b829.9-10027231)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 15.0)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Build 15A240d
    • CocoaPods version 1.13.0

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 2022.3)
    • Android Studio at /Applications/Android Studio Giraffe.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 17.0.6+0-17.0.6b829.9-10027231)

[✓] Android Studio (version 2022.2)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 17.0.6+0-17.0.6b802.4-9586694)

[✓] VS Code (version 1.83.1)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.74.0

[✓] Connected device (3 available)
    • RMX2001 (mobile) • EUYTFEUSQSRGDA6D • android-arm64  • Android 11 (API 30)
    • macOS (desktop)  • macos            • darwin-x64     • macOS 14.0 23A344 darwin-x64
    • Chrome (web)     • chrome           • web-javascript • Google Chrome 118.0.5993.70

[✓] Network resources
    • All expected network resources are available.

• No issues found!
```

```bash
[!] Flutter (Channel master, 3.16.0-13.0.pre.43, on macOS 14.0 23A344 darwin-x64, locale en-VN)
    • Flutter version 3.16.0-13.0.pre.43 on channel master at /Users/huynq/Documents/GitHub/flutter_master
    ! Warning: `flutter` on your path resolves to /Users/huynq/Documents/GitHub/flutter/bin/flutter, which is not inside your current Flutter SDK checkout at /Users/huynq/Documents/GitHub/flutter_master. Consider adding /Users/huynq/Documents/GitHub/flutter_master/bin to the front of your path.
    ! Warning: `dart` on your path resolves to /Users/huynq/Documents/GitHub/flutter/bin/dart, which is not inside your current Flutter SDK checkout at /Users/huynq/Documents/GitHub/flutter_master. Consider adding /Users/huynq/Documents/GitHub/flutter_master/bin to the front of your path.
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision c881808ed2 (50 minutes ago), 2023-10-19 02:20:19 +0000
    • Engine revision 8d51b64e94
    • Dart version 3.3.0 (build 3.3.0-41.0.dev)
    • DevTools version 2.28.1
    • If those were intentional, you can disregard the above warnings; however it is recommended to use "git" directly to perform update checks and upgrades.

[✓] Android toolchain - develop for Android devices (Android SDK version 32.0.0)
    • Android SDK at /Users/huynq/Library/Android/sdk
    • Platform android-34, build-tools 32.0.0
    • ANDROID_HOME = /Users/huynq/Library/Android/sdk
    • Java binary at: /Applications/Android Studio Giraffe.app/Contents/jbr/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 17.0.6+0-17.0.6b829.9-10027231)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 15.0)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Build 15A240d
    • CocoaPods version 1.13.0

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 2022.3)
    • Android Studio at /Applications/Android Studio Giraffe.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 17.0.6+0-17.0.6b829.9-10027231)

[✓] Android Studio (version 2022.2)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 17.0.6+0-17.0.6b802.4-9586694)

[✓] VS Code (version 1.83.1)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.74.0

[✓] Connected device (3 available)
    • RMX2001 (mobile) • EUYTFEUSQSRGDA6D • android-arm64  • Android 11 (API 30)
    • macOS (desktop)  • macos            • darwin-x64     • macOS 14.0 23A344 darwin-x64
    • Chrome (web)     • chrome           • web-javascript • Google Chrome 118.0.5993.70

[✓] Network resources
    • All expected network resources are available.

! Doctor found issues in 1 category.
```
</details>

---

Through the summary table above, we are seeing 2 mouse events (scroll & drag). Drag events can potentially work with custom scroll behavior. However, I'm also curious to know if there is any desktop/web applications with similar behavior have been implemented out there.

Also, trackpad should be addressed as well but it might be another issue.
